docs/repo-guides/release-nnf-sw/readme.md

```diff
@@ -0,0 +1,161 @@
+   not unheard of. Tread carefully if there are conflicts.
+
+    ```shell
+    git merge master
+    ```
+
+4. Verify that there are no differences between your branch and the master branch:
+
+    ```shell
+    git diff master
+    ```
+
+    If there are any differences, they must be trivial. Some READMEs may have extra lines at the
+    end.
+
+5. For `lustre-csi-driver` and `lustre-fs-operator`, there are additional files that need to track
+   the version number as well.
+
+    a. For `lustre-fs-operator`, update `config/manager/kustomization.yaml` with the correct
+    version.
+
+    b. For `lustre-csi-driver`, update `deploy/kubernetes/base/kustomization.yaml` and
+    `charts/lustre-csi-driver/values.yaml` with the correct version.
+
+6. Create a Pull Request from your branch and **target the release branch**. When merging the Pull
+   Request, **you must use a Merge Commit. Do not Rebase or Squash.**
+
+7. Once merged, update the release branch locally and then create an annotated tag:
+
+    ```shell
+    git checkout releases/v0
+    git tag -a v0.0.3 -m "Release v0.0.3"
+    git push origin --tags
+    ```
+
+8. Now that there is a tag, a release can be created via the [GitHub CLI](https://cli.github.com/).
+   Alternatively, use the [Web UI](https://github.com/NearNodeFlash/nnf-dm/releases/new).
+
+    ```bash
+    gh release create --generate-notes --verify-tag -p v0.0.3 -t "Release v0.0.3"
+    ```
+
+9. Repeat this process for each remaining component.
+
+## Release `nnf-deploy`
+
+Once the individual components are released, we need to update the submodules and
+`config/repositories.yaml` in the **master** branch before we start on the release branch. This makes
+sure that everything is now current on master.
+
+1. Update the submodules on master:
+
+    ```shell
+    git checkout master
+    git pull
+    git checkout -b update-submodules
+    git submodule foreach git checkout master
+    git submodule foreach git pull
+    ```
+
+2. Update `config/reponsitories.yaml` and update the version for `lustre-csi-driver` and `lustre-fs-operator`.
+
+3. Commit the changes and open a Pull Request against the `master` branch.
+
+4. Once merged, follow steps 1-3 from the previous section to update the release branch with master.
+
+5. There will be conflicts on the submodules after step 3. This is expected. We will update the
+   submodules to the new tags and then commit the changes.  If each tag was committed properly, the
+   follow command can do this for you:
+
+    ```shell
+    git submodule foreach 'git checkout `git describe --match="v*" HEAD`'
+    ```
+
+    Verify that each submodule is now at the proper tagged version.
+
+    ```shell
+    git submodule foreach git status
+    ```
+
+6. Do a `git add` for each of the submodules.
+
+7. Verify that `git status` is happy with `nnf-deploy` and then finalize the merge from master by
```

Comment:
Try a 'make'.  For the nnf-deploy v0.0.2 release I had to do a 'go mod tidy'.  Then the updates for go.sum and go.mod should have been added to this release.